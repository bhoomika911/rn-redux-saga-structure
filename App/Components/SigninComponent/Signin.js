import React, { Component } from 'react';
import {
  TextInput,
  Text,
  StyleSheet,
  View,
  Dimensions,
  TouchableHighlight,
} from 'react-native';

let window = Dimensions.get('window');

export default class Signin extends Component {
  constructor() {
    super();

    this.state = {
      email: '',
      password: '',
      signinDone: false,
      test: 0
    };
  }

  addTest(a) {
    return a + 10;
  }

  testUpdateState() {
    this.setState({
      test: 200
    })
  }

  onChangeEmail(email) {
    this.setState({
      email: email,
    });
  }

  onChangePassword(password) {
    this.setState({
      password: password,
    });
  }

  onPressBtn() {
    let { email, password } = this.state;
    if (email == "a@gmail.com" && password == "123456") {
      this.setState({
        signinDone: true
      });
    }


    // let { email, password } = this.state;
    // let res = callWebservice();
    // alert("signin successfully");
  }

  callWebservice() {
    return fetch('https://facebook.github.io/react-native/movies.json')
      .then((response) => response.json())
      .then((responseJson) => {
        return responseJson;
      })
      .catch((error) => {
        console.error(error);
      });
  }

  render() {
    let { email, password } = this.state;

    return (
      <View style={styles.container}>
        <View style={styles.txtInputParent}>
          <TextInput
            style={styles.txtInput}
            onChangeText={email => this.onChangeEmail(email)}
            value={email}
            placeholder="Email"
            testID="email"
          />
        </View>
        <View style={styles.txtInputParent}>
          <TextInput
            style={styles.txtInput}
            onChangeText={password => this.onChangePassword(password)}
            value={password}
            placeholder="Password"
            testID="password"
          />
        </View>
        <TouchableHighlight
          underlayColor="transparent"
          testID="singinBtnTouchable"
          onPress={() => this.onPressBtn()}>
          <View style={styles.signinBtn} testID="singinBtn">
            <Text style={styles.signinText}>Signin</Text>
          </View>
        </TouchableHighlight>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    justifyContent: 'center',
    // backgroundColor: 'red',
  },
  txtInputParent: {
    borderBottomColor: 'gray',
    borderBottomWidth: 1,
    height: 40,
    marginVertical: 10,
    width: window.width - 40,
  },
  textInput: {
    height: 40,
    width: window.width - 40,
    // paddingTop: 20,
  },
  signinBtn: {
    height: 40,
    width: window.width - 40,
    borderWidth: 1,
    backgroundColor: 'blue',
    borderRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
  signinText: {
    color: 'white',
    fontSize: 20,
  },
});
