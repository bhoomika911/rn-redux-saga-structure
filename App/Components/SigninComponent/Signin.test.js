import React from 'react';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Signin from './Signin';

configure({ adapter: new Adapter() });

global.fetch = jest.fn().mockImplementation(() => {
    let pr = new Promise((resolve, reject) => {
        resolve({
            json: () => {
                return new Promise((resolve, reject) => {
                    resolve({
                        status: "success"
                    });
                });
            }
        });
    });

    return pr;
});


describe("Unit testing : Signin page", () => {
    describe("Rendering", () => {
        test("Snapshot testing", () => {
            let wrapper = render(<Signin />);
            expect(wrapper).toMatchSnapshot();
        });

        test("find element : email text input", () => {
            let wrapper = shallow(<Signin />);
            expect(wrapper.find({ testID: "email" })).toHaveLength(1);
        });

        test("find element : password text input", () => {
            let wrapper = shallow(<Signin />);
            expect(wrapper.find({ testID: "password" })).toHaveLength(1);
        });

        test("find element : Signin button", () => {
            let wrapper = shallow(<Signin />);
            expect(wrapper.find({ testID: "singinBtn" })).toHaveLength(1);
        });
    });

    describe("Interaction", () => {
        test("onChangeEmail", () => {
            let wrapper = shallow(<Signin />);
            wrapper.find({ testID: "email" }).simulate('changeText', "a@gmail.com");
            expect(wrapper.state("email")).toBe('a@gmail.com');
        });

        test("onChangePassword", () => {
            let wrapper = shallow(<Signin />);
            wrapper.find({ testID: "password" }).simulate('changeText', "123456");
            expect(wrapper.state("password")).toBe('123456');
        });

        test("Signin : without entering password", () => {
            let wrapper = shallow(<Signin />);
            wrapper.find({ testID: "email" }).simulate('changeText', "a@gmail.com");
            wrapper.find({ testID: "singinBtnTouchable" }).simulate('press');
            expect(wrapper.state("signinDone")).toBe(false);
        });

        test("Signin : success", () => {
            let wrapper = shallow(<Signin />);
            wrapper.find({ testID: "email" }).simulate('changeText', "a@gmail.com");
            wrapper.find({ testID: "password" }).simulate('changeText', "123456");
            wrapper.find({ testID: "singinBtnTouchable" }).simulate('press');
            expect(wrapper.state("signinDone")).toBe(true);
        });

        test("callWebservice function", async () => {
            let wrapper = shallow(<Signin />).instance();
            let res = await wrapper.callWebservice();
            expect(res).toMatchObject({
                status: "success"
            });
        });

        test("testUpdateState function", () => {
            let wrapper = shallow(<Signin />);
            let instance = wrapper.instance();
            instance.testUpdateState();
            expect(wrapper.state("test")).toEqual(200);
        });
    });
});