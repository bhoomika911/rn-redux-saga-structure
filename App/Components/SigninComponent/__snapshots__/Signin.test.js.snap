// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit testing : Signin page Rendering Snapshot testing 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "flex:1;padding:20px;justify-content:center",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "style": "border-bottom-color:gray;border-bottom-width:1px;height:40px;margin-vertical:10px;width:710px",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "placeholder": "Email",
              "testid": "email",
              "underlinecolorandroid": "transparent",
              "value": "",
            },
            "children": Array [],
            "name": "textinput",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "placeholder": undefined,
              "testid": undefined,
              "underlinecolorandroid": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "placeholder": undefined,
              "testid": undefined,
              "underlinecolorandroid": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "view",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "style": "border-bottom-color:gray;border-bottom-width:1px;height:40px;margin-vertical:10px;width:710px",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "placeholder": "Password",
                "testid": "password",
                "underlinecolorandroid": "transparent",
                "value": "",
              },
              "children": Array [],
              "name": "textinput",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "placeholder": undefined,
                "testid": undefined,
                "underlinecolorandroid": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "placeholder": undefined,
                "testid": undefined,
                "underlinecolorandroid": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "view",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "focusable": "true",
              "testid": "singinBtnTouchable",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "style": "height:40px;width:710px;border-width:1px;background-color:blue;border-radius:20px;align-items:center;justify-content:center",
                  "testid": "singinBtn",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "style": "color:white;font-size:20px",
                    },
                    "children": Array [
                      Object {
                        "data": "Signin",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "text",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "style": undefined,
                    },
                  },
                ],
                "name": "view",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "style": undefined,
                  "testid": undefined,
                },
                "x-attribsPrefix": Object {
                  "style": undefined,
                  "testid": undefined,
                },
              },
            ],
            "name": "view",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "focusable": undefined,
              "testid": undefined,
            },
            "x-attribsPrefix": Object {
              "focusable": undefined,
              "testid": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {
          "style": "border-bottom-color:gray;border-bottom-width:1px;height:40px;margin-vertical:10px;width:710px",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "placeholder": "Password",
              "testid": "password",
              "underlinecolorandroid": "transparent",
              "value": "",
            },
            "children": Array [],
            "name": "textinput",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "placeholder": undefined,
              "testid": undefined,
              "underlinecolorandroid": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "placeholder": undefined,
              "testid": undefined,
              "underlinecolorandroid": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "view",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "focusable": "true",
            "testid": "singinBtnTouchable",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "style": "height:40px;width:710px;border-width:1px;background-color:blue;border-radius:20px;align-items:center;justify-content:center",
                "testid": "singinBtn",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "style": "color:white;font-size:20px",
                  },
                  "children": Array [
                    Object {
                      "data": "Signin",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "text",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "style": undefined,
                  },
                },
              ],
              "name": "view",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "style": undefined,
                "testid": undefined,
              },
              "x-attribsPrefix": Object {
                "style": undefined,
                "testid": undefined,
              },
            },
          ],
          "name": "view",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "focusable": undefined,
            "testid": undefined,
          },
          "x-attribsPrefix": Object {
            "focusable": undefined,
            "testid": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "style": "border-bottom-color:gray;border-bottom-width:1px;height:40px;margin-vertical:10px;width:710px",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "placeholder": "Email",
                "testid": "email",
                "underlinecolorandroid": "transparent",
                "value": "",
              },
              "children": Array [],
              "name": "textinput",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "placeholder": undefined,
                "testid": undefined,
                "underlinecolorandroid": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "placeholder": undefined,
                "testid": undefined,
                "underlinecolorandroid": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "view",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {
          "focusable": "true",
          "testid": "singinBtnTouchable",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "style": "height:40px;width:710px;border-width:1px;background-color:blue;border-radius:20px;align-items:center;justify-content:center",
              "testid": "singinBtn",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "style": "color:white;font-size:20px",
                },
                "children": Array [
                  Object {
                    "data": "Signin",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "text",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "style": undefined,
                },
              },
            ],
            "name": "view",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "style": undefined,
              "testid": undefined,
            },
            "x-attribsPrefix": Object {
              "style": undefined,
              "testid": undefined,
            },
          },
        ],
        "name": "view",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "style": "border-bottom-color:gray;border-bottom-width:1px;height:40px;margin-vertical:10px;width:710px",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "placeholder": "Password",
                "testid": "password",
                "underlinecolorandroid": "transparent",
                "value": "",
              },
              "children": Array [],
              "name": "textinput",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "placeholder": undefined,
                "testid": undefined,
                "underlinecolorandroid": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "placeholder": undefined,
                "testid": undefined,
                "underlinecolorandroid": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "view",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "style": "border-bottom-color:gray;border-bottom-width:1px;height:40px;margin-vertical:10px;width:710px",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "placeholder": "Email",
                  "testid": "email",
                  "underlinecolorandroid": "transparent",
                  "value": "",
                },
                "children": Array [],
                "name": "textinput",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "placeholder": undefined,
                  "testid": undefined,
                  "underlinecolorandroid": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "placeholder": undefined,
                  "testid": undefined,
                  "underlinecolorandroid": undefined,
                  "value": undefined,
                },
              },
            ],
            "name": "view",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "style": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "focusable": undefined,
          "testid": undefined,
        },
        "x-attribsPrefix": Object {
          "focusable": undefined,
          "testid": undefined,
        },
      },
    ],
    "name": "view",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
